ecosystem/typescript/sdk_v2/src/core/account.ts

```diff
@@ -0,0 +1,106 @@
+import nacl from "tweetnacl";
+import * as bip39 from "@scure/bip39";
+import { bytesToHex } from "@noble/hashes/utils";
+import { derivePath, Memoize } from "../utils";
+
+export class Account {
+  /**
+   * A private key and public key, associated with the given account
+   */
+
+  /**
+   * Address associated with the given account
+   */
+
+  /**
+   * Creates new account instance.
+   * @param privateKeyBytes  Private key from which account key pair will be generated.
+   * If not specified, new key pair is going to be created.
+   */
+  constructor(private signingKey: nacl.SignKeyPair, private accountAddress: AccountAddress) {}
+
+  public get publicKey(): Uint8Array {
+    return this.signingKey.publicKey;
+  }
+
+  public get privateKey(): Uint8Array {
+    return this.signingKey.secretKey;
+  }
+
+  public get address(): string {
+    return this.accountAddress;
+  }
+
+  static generateFromPrivateKey(privateKeyBytes: Uint8Array) {
+    const signingKey = nacl.sign.keyPair.fromSeed(privateKeyBytes.slice(0, 32));
+    const accountAddress = new AccountAddress(Account.authKey(signingKey.publicKey).toBytes());
+
+    return new Account(signingKey, accountAddress);
+  }
+
+  static generateFromPrivateKeyAndAddress(privateKeyBytes: Uint8Array, address: AccountAddress) {
+    const accountAddress = new AccountAddress(Hex.validate(address).toBytes());
+    const signingKey = nacl.sign.keyPair.fromSeed(privateKeyBytes.slice(0, 32));
+
+    return new Account(signingKey, accountAddress);
+  }
+
+  static generate() {
+    const signingKey = nacl.sign.keyPair();
+    const accountAddress = new AccountAddress(Account.authKey(signingKey.publicKey).toBytes());
+
+    return new Account(signingKey, accountAddress);
+  }
+
+  @Memoize()
+  static authKey(publicKey: Uint8Array): Hex {
+    const pubKey = new Ed25519PublicKey(publicKey);
+    const authKey = AuthenticationKey.fromEd25519PublicKey(pubKey);
+    return authKey.derivedAddress();
+  }
+
+  /**
+   * Creates new account with bip44 path and mnemonics,
+   * @param path. (e.g. m/44'/637'/0'/0'/0')
```

Comment:
This is the default that the wallets use right? Or does each wallet use a different derivation path?

I remember when I used to import my accounts by seed phrase I wouldn't understand why the accounts created were different and it was cause of the derivation path, would be nice if we could make it clear

Reply:
Can do!

Btw this is a draft, I am gonna re-do the whole thing 😄

Reply:
Gotcha :D